src/__test__/test-utils/vega-utils.js

```diff
@@ -0,0 +1,23 @@
+import * as rtl from '@testing-library/react';
+
+const expectStringInVegaCanvas = async (str, ocurrences, waitForOptions) => {
+  const getCanvas = () => rtl.screen.getByRole('graphics-document').children[0];
+  // eslint-disable-next-line no-underscore-dangle
+  const getCanvasStrings = (canvas) => canvas.getContext('2d').__getEvents()
+    .filter((event) => event.type === 'fillText')
+    .map((event) => event.props.text);
+  const canvas = getCanvas();
+  const onTimeout = (error) => {
+    console.log(`Could not find ${ocurrences} ocurrences of "${str}" in the canvas. I found these: ${getCanvasStrings(canvas)}`);
+    return error;
+  };
+  await rtl.waitFor(() => {
+    const strings = getCanvasStrings(canvas);
+    expect(strings.filter((inCanvas) => inCanvas.includes(str)).length).toBe(ocurrences);
+  }, { ...waitForOptions, onTimeout });
+};
```

Comment:
This means that the test will still error out on timeout, but returns a better dump than previously right?

Reply:
@aerlaut The key to prevent the timeout is to use a one letter input for the title , so that throttling prevention does not kick in and causes additional updates. You are correct, though: we'll get more useful and specific logging if the timeout reappears.